Improve Github Issue Templates

travelton opened the pull request:
- Improved Github Issue templates to ensure we are capturing information consistently between issue types.
- Fixed formatting of labels, to ensure labels are automatically applied.
- Updated a few links to new domain.

travelton committed:
Improve Github Issue Templates

webvictim commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -10,22 +10,43 @@ labels: "type: bug"
+### Debug Logs
+Please include or attach debug logs, when appropriate. Obfuscate sensitive information!
+- Start Teleport with --debug flag (`teleport --debug`)
+- Run Tsh with --debug flag (`tsh --debug`)
```
```suggestion
- Run tsh with --debug flag (`tsh --debug`)
```

webvictim commented on .github/ISSUE_TEMPLATE/documentation.md:
```diff
@@ -1,14 +1,27 @@
 name: Documentation 📝
-about: Suggest better documentation for a particular tool or process. 
-labels: "type: documentation"
+about: Suggest improvements to Teleport's documentation site.
```
```suggestion
about: Suggest improvements to Teleport's documentation.
```

webvictim commented on .github/ISSUE_TEMPLATE/question.md:
```diff
@@ -1,26 +1,34 @@
-  - For Pro and Enterprise users, please open a ticket via the Enterprise support portal: https://dashboard.gravitational.com/web/
-
-  Before opening a new issue, please search existing issues: https://github.com/gravitational/teleport/issues
+  - For Open Source users, please ask in our community forum: https://community.goteleport.com/c/teleport
+  - For Enterprise customers, please open a support ticket via the Enterprise support portal: https://dashboard.gravitational.com/web/
```
Given that we're giving these two options, is there any point in having this issue type at all?

travelton replied:
In `config.yml`, perhaps we add a support option? Then eliminate `question.md`. I've only seen it used once. 

- Teleport Documentation
- Teleport Community Forum
- **Teleport Enterprise Support** -> https://dashboard.gravitational.com/web/
- Request a Demo

webvictim replied:
Great idea!

benarent commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -10,22 +10,43 @@ labels: "type: bug"
+- [x] Teleport Enterprise
+- [ ] Other
+
+### Which component(s)
```
This is handy.

benarent reviewed (approved):
Looks good, I just merged other updates earlier today that cleaned up some Gravitational vs GoTeleport issues.

travelton committed:
Merge branch 'master' into travis/github-templates

travelton committed:
PR feedback.

- Fixed tsh case in bug_report.md
- Removed `site` from about for documentation.md
- Cap case Community Forum in config, so it matches the other options in the list.

russjones commented on .github/ISSUE_TEMPLATE/config.yml:
```diff
@@ -3,7 +3,7 @@ contact_links:
     about: Check out the Teleport documentation for answers to common questions
-  - name: Teleport community forum
+  - name: Teleport Community Forum
     url: https://community.goteleport.com/
```
Update to GitHub discussions.

russjones reviewed (approved):
@travelton One comment, then please merge.

travelton committed:
Merge branch 'master' into travis/github-templates

travelton committed:
Update community to Github Discussions.

travelton committed:
Update Question template to Github Discussions.

benarent commented:
In hindsight the check list is making the issues look pretty strange? I wonder if we can keep something but not bubble it up to the Github list? 

![image](https://user-images.githubusercontent.com/559288/108277346-22ba4880-712e-11eb-962f-fdc5372265ae.png)

travelton commented:
@benarent That's annoying. I don't see any way around this with task lists. 

What if we just include a list, instead of a task list. Then, the user can delete items that are not applicable. 

```
### Which platform(s)
- Teleport Open Source
- Teleport Cloud
- Teleport Enterprise
- Other

### Which component(s)
- Server Access
- Kubernetes Access
- Application Access
- Database Access
- CLI Clients
- Web Clients
- Other
```

benarent commented:
^ Yeah, I think that's a good idea. You can add a markdown comment that'll let people know
